vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -59,26 +65,419 @@ func extractConversionMetrics(cve cves.CVE5, refs []osvschema.Reference) {
+// events to an OSV version range. If 'intro' is empty, it defaults to "0".
+func addToVersionRange(versionRange *osvschema.Range, intro string, lastAff string, fixed string) {
+	var i string
+	if intro == "" {
+		i = "0"
+	} else {
+		i = intro
+	}
+	versionRange.Events = append(versionRange.Events, osvschema.Event{
+		Introduced: i})
+
+	if fixed != "" {
+		versionRange.Events = append(versionRange.Events, osvschema.Event{
+			Fixed: fixed})
+	} else if lastAff != "" {
+		versionRange.Events = append(versionRange.Events, osvschema.Event{
+			LastAffected: lastAff,
+		})
+	}
+}
+
+// findCPEVersionRanges extracts version ranges and CPE strings from the CNA's
+// CPE applicability statements in a CVE record.
+func findCPEVersionRanges(cve cves.CVE5) (versionRanges []osvschema.Range, cpes []string, err error) {
+	// TODO(jesslowe): Add logic to also extract CPEs from the 'affected' field (e.g., CVE-2025-1110).
+	for _, c := range cve.Containers.CNA.CPEApplicability {
+		for _, node := range c.Nodes {
+			if node.Operator != "OR" {
+				continue
+			}
+			for _, match := range node.CPEMatch {
+				if !match.Vulnerable {
+					continue
+				}
+				cpes = append(cpes, match.Criteria)
+				versionRange := osvschema.Range{}
+
+				// If no start version is given, assume the vulnerability starts from version "0".
+				if match.VersionStartIncluding == "" {
+					match.VersionStartIncluding = "0"
+				}
+
+				if match.VersionEndExcluding != "" {
+					addToVersionRange(&versionRange, match.VersionStartIncluding, "", match.VersionEndExcluding)
+				} else if match.VersionEndIncluding != "" {
+					addToVersionRange(&versionRange, match.VersionStartIncluding, match.VersionEndIncluding, "")
+				}
+				if len(versionRange.Events) != 0 {
+					versionRanges = append(versionRanges, versionRange)
+				}
+			}
+		}
+	}
+	if len(versionRanges) == 0 {
+		return nil, nil, fmt.Errorf("no versions extracted from CPEs")
+	}
+	return versionRanges, cpes, nil
 }
 
-// FromCVE5 creates a Vulnerability from a CVE5 object.
+// ExtractVersionsFromAffectedField extracts version ranges from a CVE 'affected' entry.
+// It handles various scenarios:
+// - If `defaultStatus` is "affected", it may calculate the inverse ranges (common for Linux).
+// - It iterates through `versions` entries with `status: "affected"`.
+// - It constructs ranges from `version`, `lessThan`, and `lessThanOrEqual` fields.
+// - For GitHub CVEs, it uses a special parser for version strings like "< 1.2.3".
+// - For git commits, it creates an introduced event.
+// - As a fallback, it may assume a single version means "fixed at this version, introduced at 0".
+//
+// Returns the extracted OSV ranges, the most frequent version type (e.g., "semver"), and any notes.
+func ExtractVersionsFromAffectedField(affected cves.Affected, cnaAssigner string) (versionRanges []osvschema.Range, rangeType string, notes []string) {
```

Comment:
Should this be a private function? To match the other extractVersions functions. (or make them all public and move it to a separate package).